src/main/java/ca/corefacility/bioinformatics/irida/ria/web/linelist/LineListPermissions.java

```diff
@@ -0,0 +1,72 @@
+package ca.corefacility.bioinformatics.irida.ria.web.linelist;
+
+import java.security.Principal;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+
+import ca.corefacility.bioinformatics.irida.model.project.Project;
+import ca.corefacility.bioinformatics.irida.model.sample.Sample;
+import ca.corefacility.bioinformatics.irida.model.user.Role;
+import ca.corefacility.bioinformatics.irida.model.user.User;
+import ca.corefacility.bioinformatics.irida.security.permissions.project.ProjectOwnerPermission;
+import ca.corefacility.bioinformatics.irida.security.permissions.sample.UpdateSamplePermission;
+import ca.corefacility.bioinformatics.irida.service.user.UserService;
+
+/**
+ * Component to standardize user permissions for samples on the
+ * linelist page.
+ */
+@Component
+public class LineListPermissions {
+	private UserService userService;
+	private UpdateSamplePermission updateSamplePermission;
+	private ProjectOwnerPermission projectOwnerPermission;
+
+	//
+	private boolean isAdmin;
+	private Authentication authentication;
+
+	@Autowired
+	public LineListPermissions(UserService userService, UpdateSamplePermission updateSamplePermission,
+			ProjectOwnerPermission projectOwnerPermission) {
+		this.userService = userService;
+		this.updateSamplePermission = updateSamplePermission;
+		this.projectOwnerPermission = projectOwnerPermission;
+	}
+
+	/**
+	 * Set the {@link Principal} user to create the permissions.
+	 *
+	 * @param principal {@link Principal}
+	 */
+	public void setPrincipalPermissions(Principal principal) {
+		User user = userService.getUserByUsername(principal.getName());
+		this.isAdmin = user.getSystemRole()
+				.equals(Role.ROLE_ADMIN);
+		this.authentication = SecurityContextHolder.getContext()
```

Comment:
Yeah don't do this please.  This method should likely not be in here.  The `LineListPermissions` object is shared between users in the application so the `Authentication` and `isAdmin` props could leak between users.

Reply:
That is actually why I made the LineListController session scoped.